restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -1710,15 +1750,35 @@ private void outbound(final Object message, final ActorRef sender) throws Servle
         ec.executePreOutboundAction(request, this.extensions);
         switch (request.type()) {
             case CLIENT: {
-                if (request.isAllowed()) {
-                    outboundToClient(request, sender);
+                ClientsDao clients = storage.getClientsDao();
+                Client client = clients.getClient(request.to().replaceFirst("client:", ""));
+                if (client != null) {
+                    long delay = sendPushNotificationIfNeeded(client.getPushClientIdentity());
+                    system.scheduler().scheduleOnce(Duration.create(delay, TimeUnit.MILLISECONDS), new Runnable() {
+                        @Override
+                        public void run() {
+                            createCallRequest = request;
+                            try {
+                                if (request.isAllowed()) {
+                                    outboundToClient(request, sender);
+                                } else {
+                                    //Extensions didn't allowed this call
+                                    final String errMsg = "Not Allowed to make this outbound call";
+                                    logger.warning(errMsg);
+                                    sender.tell(new CallManagerResponse<ActorRef>(new RestcommExtensionException(errMsg), createCallRequest), self());
+                                }
+                                ExtensionController.getInstance().executePostOutboundAction(request, extensions);
+                            } catch (ServletParseException e) {
+                                throw new RuntimeException(e);
+                            }
+                        }
+                    }, system.dispatcher());
```

Comment:
@agafox see my previous answer and check here http://doc.akka.io/docs/akka/2.5.3/scala/dispatchers.html#blocking-needs-careful-management if you need more details on the problem